Implement transitional crown favicons (v5.0) (#4445)
alphagov/govuk-frontend: CHANGELOG.md

## Unreleased

### Breaking changes

> #### Check your favicons, app icons and OpenGraph image still work

Reviewer: I think make 'image' plural instead of 'work'.

> #### Check your favicons, app icons and OpenGraph image still work

Reviewer 2: My brain said to do this but there's only a single OpenGraph image

Would you make _images_ apply the other types too?

```suggestion
#### Check your favicon, app icon and OpenGraph images still work
```

> #### Check your favicons, app icons and OpenGraph image still work

Reviewer: Ah ok, another option is to put 'OpenGraph image' at the start of the list, the brain ignores the inconsistency

> We've changed the names, formats and sizes of icon assets that we distribute in Frontend. You will want to check that the correct files are in place.

Reviewer 2: ```suggestion
We've changed the names, formats and sizes of icon assets that we distribute in GOV.UK Frontend. You will want to check that the correct files are in place.
```

Reviewer: I made a couple of tweaks to Colin's suggestion

> The following files have been added to the assets folder:

Reviewer 2: I think we need to call out `assetPath` (which defaults to `/assets`), `assetUrl` and `opengraphImageUrl`

Could be helpful to refer to the link in https://github.com/alphagov/govuk-frontend/pull/4445/files#r1389317268 earlier but see what @claireashworth recommends

Reviewer: Can do this. Another option is to move the file path line up between current lines 11 and 13, as it's an action users need to do

Reviewer 2: Was thinking more for context as lots of services use project-specific paths. When you read "assets folder" it doesn't necessarily help identify where the images are copied or deployed

>If a user is already using the Nunjucks template (and thus `assetPath`, et al.), then they are will already have these variables configured. They don't need to be changed if the images are being replaced in-place.

Even the Nunjucks templates can override them elsewhere:

```mjs
{
  assetPath: "/specific/directory/that/other/team/manages",
  assetUrl: "http://cdn-assets-here.domain.com/assets",
  opengraphImageUrl: "http://social-media-contract-assets-here.domain.com/assets/specific.png"
}
```

Author: Do we still think specifically calling it out is needed now that documentation relating to it is linked from the changelog entry?

Reviewer: I think it should be ok as is. We can always amend it if we get feedback to the contrary.

Reviewer 2: Nice, thanks for sticking the link in. Yeah we're all set now 🙌

> The following files have been added to the assets folder:

Reviewer: ```suggestion
We've added the following files to the assets folder:
```

- manifest.json
- images/favicon.svg
- images/govuk-icon-180.png
- images/govuk-icon-192.png
- images/govuk-icon-512.png
- images/govuk-icon-mask.svg

The following files have been removed from the assets folder:

- images/govuk-apple-touch-icon.png
- images/govuk-apple-touch-icon-152x152.png
- images/govuk-apple-touch-icon-167x167.png
- images/govuk-apple-touch-icon-180x180.png
- images/govuk-mask-icon.svg

If you're not using the Nunjucks page template, you will need to replace the list of icons in the template's `head` with the following:

```html
<link rel="icon" sizes="48x48" href="/assets/images/favicon.ico">
<link rel="icon" sizes="any" href="/assets/images/favicon.svg" type="image/svg+xml">
<link rel="mask-icon" href="/assets/images/govuk-icon-mask.svg" color="#0b0c0c">
<link rel="apple-touch-icon" href="/assets/images/govuk-icon-180.png">
<link rel="manifest" href="/assets/manifest.json">
```

> You will need to update the file path to match your assets folder if it's not at the default location.

Reviewer 2: I've pinched this from **Remove the fallback GOV.UK crown logo from your HTML**

Think it's helpful?

```suggestion
You will need to update the file paths to match your assets folder depending on [how you are serving the font and image assets](https://frontend.design-system.service.gov.uk/importing-css-assets-and-javascript/#font-and-image-assets).
```
